bundles/org.openhab.binding.vigicrues/src/main/java/org/openhab/binding/vigicrues/internal/handler/VigiCruesHandler.java

```diff
@@ -133,8 +258,29 @@ private void updateQuantity(String channelId, Double value, Unit<?> unit) {
 
     public void updateDate(String channelId, ZonedDateTime zonedDateTime) {
         if (isLinked(channelId)) {
-            ZonedDateTime localDateTime = zonedDateTime.withZoneSameInstant(timeZoneProvider.getTimeZone());
-            updateState(channelId, new DateTimeType(localDateTime));
+            updateState(channelId, new DateTimeType(zonedDateTime));
+        }
+    }
+
+    public void updateAlert(String channelId, int value) {
+        String channelIcon = channelId + "-icon";
+        if (isLinked(channelId)) {
+            updateState(channelId, new DecimalType(value));
+        }
+        if (isLinked(channelIcon)) {
+            String resource = getResource(String.format("picto/crue-%d.svg", value));
+            updateState(channelIcon,
+                    resource != null ? new RawType(resource.getBytes(), "image/svg+xml") : UnDefType.UNDEF);
+        }
+    }
+
+    public @Nullable String getResource(String iconPath) {
+        Bundle bundle = FrameworkUtil.getBundle(getClass());
+        try (InputStream stream = bundle.getResource(iconPath).openStream()) {
+            return new BufferedReader(new InputStreamReader(stream)).lines().collect(Collectors.joining("\n"));
```

Comment:
If you want to just return a binary array this can be vastly simplified due to java 11.
```suggestion
            return stream.readAllBytes();
```